### PR TITLE
Search: restore old syntax for repo:contains filters

### PR DESCRIPTION
**Pull request by jtibshirani**
In 4.0, we renamed the `repo.contains` predicates to be more consistent. At the
time, we didn't think any users would be affected, so we cut over fully to the
new syntax. But some users have scripts or batch changes that use the old
syntax, which will break on 4.0 with no grace period to cut over.

This change restores the old syntax alongside the new, to give users a time
period to transition. We still plan to remove the old syntax in the future. The
old syntax is only restored in the backend, it's not documented or part of
autocomplete.

Partial revert of #40389

## Test plan

Restored relevant unit tests. Manually tested using `sg start` -- old syntax works, but is not documented.

### CODE REVIEW

**Comment by jtibshirani** [comment]
I'm not sure why sync-tracking-issues often fails for me 🤔

File: internal/search/query/predicate.go [hunk 1]
```diff
@@ -119,19 +123,41 @@ func (f *RepoContainsFilePredicate) Unmarshal(params string, negated bool) error
 		return err
 	}
 
-	for _, node := range nodes {
-		if err := f.parseNode(node); err != nil {
+	if err := f.parseNodes(nodes); err != nil {
+		// If there's a parsing error, try falling back to the deprecated syntax `repo:contains.file(name.go)`.
+		// Only attempt to fall back if there is a single pattern node, to avoid being too lenient.
```
Comment (camdencheek) [inline internal/search/query/predicate.go:128]: This is a nice touch. Thanks for the effort to make this "just work"

File: internal/search/query/predicate.go [hunk 2]
```diff
@@ -318,6 +344,78 @@ func (p *RepoHasKeyPredicate) Unmarshal(params string, negated bool) (err error)
 func (p *RepoHasKeyPredicate) Field() string { return FieldRepo }
 func (p *RepoHasKeyPredicate) Name() string  { return "has.key" }
 
+// RepoContainsPredicate represents the `repo:contains(file:a content:b)` predicate.
+// DEPRECATED: this syntax is deprecated in favor of `repo.contains.file`.
```
Comment (camdencheek) [inline internal/search/query/predicate.go:348]:
```suggestion
// DEPRECATED: this syntax is deprecated in favor of `repo:contains.file`.
```

File: internal/search/query/predicate.go [hunk 3]
```diff
@@ -318,6 +344,78 @@ func (p *RepoHasKeyPredicate) Unmarshal(params string, negated bool) (err error)
...
+	}
+	f.Negated = negated
+	return nil
+}
+
+func (f *RepoContainsPredicate) parseNode(n Node) error {
+	switch v := n.(type) {
+	case Parameter:
+		if v.Negated {
+			return errors.New("predicates do not currently support negated values")
```
Comment (camdencheek) [inline internal/search/query/predicate.go:377]:
```suggestion
			return errors.New("the repo:contains() predicate does not currently support negated arguments")
```

File: internal/search/query/predicate.go [hunk 4]
```diff
@@ -318,6 +344,78 @@ func (p *RepoHasKeyPredicate) Unmarshal(params string, negated bool) (err error)
...
+		}
+		switch strings.ToLower(v.Field) {
+		case "file":
+			if f.File != "" {
+				return errors.New("cannot specify file multiple times")
+			}
+			if _, err := regexp.Compile(v.Value); err != nil {
+				return errors.Errorf("`contains` predicate has invalid `file` argument: %w", err)
```
Comment (camdencheek) [inline internal/search/query/predicate.go:385]: It's not clear to the user that `contains` is scoped to the `repo:` filter here

```suggestion
				return errors.Errorf("the `repo:contains()` predicate has invalid `file` argument: %w", err)
```

File: internal/search/query/predicate.go [hunk 5]
```diff
@@ -318,6 +344,78 @@ func (p *RepoHasKeyPredicate) Unmarshal(params string, negated bool) (err error)
...
+			}
+			f.File = v.Value
+		case "content":
+			if f.Content != "" {
+				return errors.New("cannot specify content multiple times")
+			}
+			if _, err := regexp.Compile(v.Value); err != nil {
+				return errors.Errorf("`contains` predicate has invalid `content` argument: %w", err)
```
Comment (camdencheek) [inline internal/search/query/predicate.go:393]:
```suggestion
				return errors.Errorf("the `repo:contains()` predicate has invalid `content` argument: %w", err)
```

**Review by camdencheek — APPROVED** [review]
I completely forgot that we changed the `repo:contains.file()` behavior too. Thanks for taking care of this!

**Comment by camdencheek** [comment]
`sync-tracking-issues` has always been flaky. It's always safe to ignore

**Pull request merged**